### DISCUSSION TITLE
ACCESS VIOLATION no .AddFile

### DISCUSSION THREAD

**Discussion by DannyahIA** [viniciussanchez/RESTRequest4Delphi#189] [Q&A] [answered]
Boa Tarde estou tendo um problema com o envio de arquivos, onde recebo um ACCESS VIOLATION.

O código que estou usando é o seguinte, ele comprime um .exe e depois o envia para o server REST :

```
procedure CompressFile(appPath: string; appInfo: TApplicationInfo);
var
  Zip: TZipper;
  Streamer: TMemoryStream;
  Response: IResponse;
begin
  Zip := TZipper.Create;
  Streamer := TMemoryStream.Create;
  try
    if not FileExists(appPath) then
      raise EFileNotFoundException.Create('File ' + appPath + ' does not exist.');

    Zip.Entries.AddFileEntry(appPath, ExtractFileName(appPath));
    Zip.SaveToStream(Streamer);
    Streamer.Position := 0;

    appInfo.NewApplicationInfo.ZipFileName :=
      Copy(ExtractFileName(appPath), 0, Length(ExtractFileName(appPath)) - 4) + '.zip';

    Response := TRequest.New.BaseURL(EnvConfig.Url + '/upload')
      .ContentType('application/multiform-data')
      .TokenBearer(EnvConfig.AuthToken)
      .AddFile('file', Streamer, appInfo.NewApplicationInfo.ZipFileName, 'multipart/form-data')
      .Post;

    ShowMessage(Response.Content);

  finally
    Zip.Free;
    Streamer.Free;
  end;
end; 
``` 

Realizei testes no meu código, e criei um novo projeto para constatar que não estava ocorrendo algum erro em outra parte do meu projeto, porem constatei que o erro persiste e ocorre durante a requisição, que nem chega no REST.

![image](https://github.com/viniciussanchez/RESTRequest4Delphi/assets/133452829/d84734a5-bdc7-4ea0-98be-faad88162e58)

estou utilizando o Lazarus como IDE e a ultima versão disponivel do RESTRequest4D.

**Comment by viniciussanchez** [comment, marked as answer]
Você quis dizer que no caso a requisição nem bate no seu endpoint na API? Se for isso, a sua API está usando o middleware octet-stream? Se estiver, tente atualizar ele por favor.
